From: Debajyoti Bera <dbera.web@gmail.com>

While migrating to Lucene.Net-1.9, it was patched numerous times to keep in sync
with the beagle specific changes to Lucene.Net. Each incremental was checked in separately.
This patch should be applied after all of them to fix some errors in the previous
patches.

Index: Store/FSDirectory.cs
===================================================================
RCS file: /cvs/gnome/beagle/beagled/Lucene.Net/Store/FSDirectory.cs,v
retrieving revision 1.19.2.18
diff -u -3 -p -r1.19.2.18 FSDirectory.cs
--- Store/FSDirectory.cs	30 Sep 2006 02:52:27 -0000	1.19.2.18
+++ Store/FSDirectory.cs	30 Sep 2006 02:56:11 -0000
@@ -121,8 +121,6 @@ namespace Lucene.Net.Store
 		    //System.IO.StreamWriter w = new System.IO.StreamWriter (new Mono.Unix.UnixStream (fd, true));
 		    //w.WriteLine (System.Diagnostics.Process.GetCurrentProcess ().Id);
 		    //w.Close ();
-                    System.IO.FileStream createdFile = lockFile.Create();
-                    createdFile.Close();
                     return true;
                 }
                 catch (Exception e)
Index: Store/Lock.cs
===================================================================
RCS file: /cvs/gnome/beagle/beagled/Lucene.Net/Store/Lock.cs,v
retrieving revision 1.6.4.12
diff -u -3 -p -r1.6.4.12 Lock.cs
--- Store/Lock.cs	30 Sep 2006 02:52:27 -0000	1.6.4.12
+++ Store/Lock.cs	30 Sep 2006 02:56:11 -0000
@@ -58,14 +58,14 @@ namespace Lucene.Net.Store
 		/// <throws>  IOException if lock wait times out or obtain() throws an IOException </throws>
 		public virtual bool Obtain(long lockWaitTimeout)
 		{
-			int maxSleepCount = (int) (lockWaitTimeout / LOCK_POLL_INTERVAL);
 			int sleepCount = 0;
-			maxSleepCount = System.Math.Max (maxSleepCount, 1);
-			FSDirectory.Log ("Lock.Obtain timeout={0} maxsleepcount={1}", lockWaitTimeout, maxSleepCount);
 			bool locked = Obtain();
+			int maxSleepCount = (int) (lockWaitTimeout / LOCK_POLL_INTERVAL);
+			maxSleepCount = System.Math.Min (maxSleepCount, 1);
 
 			while (!locked)
 			{
+				FSDirectory.Log ("Lock.Obtain timeout: sleepcount = {0} (timeout={1},maxsleepcount={2})", sleepCount, lockWaitTimeout, maxSleepCount);
 				if (sleepCount++ == maxSleepCount)
 				{
 					// Try and be a little more verbose on failure
Index: Analysis/Standard/StandardTokenizerTokenManager.cs
===================================================================
RCS file: /cvs/gnome/beagle/beagled/Lucene.Net/Analysis/Standard/StandardTokenizerTokenManager.cs,v
retrieving revision 1.7.4.10
diff -u -3 -p -r1.7.4.10 StandardTokenizerTokenManager.cs
--- Analysis/Standard/StandardTokenizerTokenManager.cs	30 Sep 2006 02:57:25 -0000	1.7.4.10
+++ Analysis/Standard/StandardTokenizerTokenManager.cs	30 Sep 2006 02:59:27 -0000
@@ -30,6 +30,7 @@ namespace Lucene.Net.Analysis.Standard
 
 		private void  InitBlock()
 		{
+		}
 		private int JjMoveStringLiteralDfa0_0()
 		{
 			return JjMoveNfa_0(0, 0);
